front/lib/actions/mcp_internal_actions/servers/slack/slack_personal.ts

```diff
@@ -847,127 +847,61 @@ async function createServer(
-    {
-      nameFilter: z
-        .string()
-        .optional()
-        .describe("The name of the channel to filter by (optional)"),
-    },
-    withToolLogging(
-      auth,
-      {
-        toolNameForMonitoring: SLACK_TOOL_LOG_NAME,
-        agentLoopContext,
-      },
-      async ({ nameFilter }, { authInfo }) => {
-        const accessToken = authInfo?.token;
-        if (!accessToken) {
-          return new Err(new MCPError("Access token not found"));
-        }
+2. SEARCH QUERY (lookup_type='search'): Use for channel names or vague queries. Searches across channel names, topics, and purpose descriptions. Returns top ${MAX_CHANNEL_SEARCH_RESULTS} matches as JSON array.
 
-        try {
-          return await executeListJoinedChannels(
-            nameFilter,
-            accessToken,
-            mcpServerId
-          );
-        } catch (error) {
-          const authError = handleSlackAuthError(error);
-          if (authError) {
-            return authError;
-          }
-          return new Err(new MCPError(`Error listing channels: ${error}`));
-        }
-      }
-    )
-  );
+SCOPE BEHAVIOR (only applies to lookup_type='search'):
+IMPORTANT: ALWAYS start with scope='joined' (the default). This searches the user's joined channels (public + private) first, then automatically falls back to all public workspace channels if no results are found.
 
-  server.tool(
-    "get_channel_details",
-    "Get detailed information about a specific channel including all metadata, properties, canvas, tabs, and settings. Use this when you need complete channel information beyond the basic fields provided by list_channels.",
+Only use scope='all' in these two cases:
+1. The user explicitly asks to search non-joined or public workspace channels
+2. You already tried scope='joined' and got no results (though the fallback handles this automatically)
+
+`,
     {
-      channel: z.string().describe("The channel ID or name to get details for"),
+      query: z
+        .string()
+        .describe(
+          "Channel ID (e.g., 'C01234ABCD'), channel name, or search keywords."
+        ),
+      lookup_type: z
+        .enum(["exact", "search"])
+        .describe(
+          "'exact' for channel ID only (not names), 'search' for names/keywords."
+        ),
+      scope: z
+        .enum(["joined", "all"])
+        .default("joined")
+        .describe(
+          "'joined' (default, always use this), 'all' (only if user explicitly requests non-joined channels)."
+        ),
     },
     withToolLogging(
       auth,
       {
         toolNameForMonitoring: SLACK_TOOL_LOG_NAME,
         agentLoopContext,
       },
-      async ({ channel }, { authInfo }) => {
+      async ({ query, lookup_type, scope }, { authInfo }) => {
         const accessToken = authInfo?.token;
         if (!accessToken) {
           return new Err(new MCPError("Access token not found"));
         }
```

Comment:
scope should be ensured to not be anything else than the default if lookup_type is exact